src/cursor/abstract_cursor.ts

```diff
@@ -572,6 +572,10 @@ export abstract class AbstractCursor extends EventEmitter {
       return;
     }
 
+    if (!this[kSession]) {
+      throw new Error('Should have a session when calling getMore');
```

Comment:
This ended up being the cause of a bug I had a very hard time tracking down. It only happens on `latest` running with `--setParameter requireApiVersion=1`. I had to make the changes below to avoid this error being thrown in a number of tests. @nbbeeken might this be related to your sessions work?

Reply:
Interesting, it could be, but its not obvious at first since it doesn't go through the client startSession method, its going directly to the topology one. If you can reproduce it can you take a look at:

https://github.com/mongodb/node-mongodb-native/blob/ee1a4d32ac95b7d143b08896bc486cfa8c2895a1/src/cursor/abstract_cursor.ts#L623-L625

This is where `this[kSession]` should be getting assigned so also printing the TopologyDescription with all the server descriptions as well could help

`util.inspect(cursor[kTopology].s.description, {depth: 6})` (6 usually goes plenty deep)

Reply:
Ok whoops, scrolled just a bit to see the changes you made to that section, that looks like the same issue to me, since you moved grabbing the session into the callback, then executeOperation has been called so server selection has run, now the function will return that sessions are supported. I am curious though what state the topology is in such that you're getting a failure, and then what state it transitions to after executeOperation runs.